Don't allow citizen to add reservation on top of absence marked by staff or on holidays

zch opened the pull request.

akheron commented:
Perhaps not a problem of this PR per se, but I find this icon a bit confusing:

<img width="443" alt="Screenshot 2022-09-28 at 12 55 34" src="https://user-images.githubusercontent.com/70927/192749706-ff611938-5316-41a3-80e5-a4bcc2f1ece2.png">

<img width="435" alt="Screenshot 2022-09-28 at 12 55 42" src="https://user-images.githubusercontent.com/70927/192749741-8c59893e-bd54-4a77-84f1-fcb946fc47ab.png">

No matter whether you're adding a reservation or changing it to an absence, the icon always has a human figure and a minus sign.

akheron reviewed (approved):
Otherwise looks good 👍

zch commented:
@akheron While working on the related ticket about preventing reservations on holidays, I noticed an issue with the layout on mobile devices. So I changed the CSS grid to a flexbox layout to make everything flow nicely and added the holiday stuff to this PR as it's related, and a small change..... Sorry :D
